eng/scripts/Cadl-Project-Sync.ps1

```diff
@@ -0,0 +1,101 @@
+[CmdletBinding()]
+param (
+    [Parameter(Position=0)]
+    [ValidateNotNullOrEmpty()]
+    [string] $ProjectDirectory
+)
+
+$ErrorActionPreference = "Stop"
+
+function AddSparseCheckoutPath([string]$subDirectory, [string]$file) {
+    if (!(Test-Path $file) -or !((Get-Content $file).Contains($subDirectory))) {
+        Write-Output $subDirectory >> .git/info/sparse-checkout
+    }
+}
+
+function NpmInstallForProject([string]$workingDirectory) {
+    Push-Location $workingDirectory
+
+    $currentDur = Resolve-Path "."
+    Write-Host "Generating from $currentDur"
+    if (Test-Path ".\package.json") {
+        Remove-Item -Path ".\package.json" -Force
+    }
+    if (Test-Path ".\.npmrc") {
+        Remove-Item -Path ".\.npmrc" -Force
+    }
+    npm install "@azure-tools/cadl-csharp"
+
+    Pop-Location
+}
+
+function CopySpecToProjectIfNeeded([string]$pullResult, [string]$specCloneRoot, [string]$mainSpecDir, [string]$dest, [string[]]$specAdditionalSubDirectories) {
+    if (!($pullResult.Contains("Already up to date.")) && Test-Path $dest) {
+
+        $source = "$specCloneRoot/$mainSpecDir"
+        Write-Host "Copying spec from $source"
+        Copy-Item -Path $source -Destination $dest -Recurse -Force
+        foreach($additionalDir in $specAdditionalSubDirectories)
+        {
+            $source = "$specCloneRoot/$additionalDir"
+            Write-Host "Copying spec from $source"
+            Copy-Item -Path $source -Destination $dest -Recurse -Force
+        }
+    }
+}
+
+function UpdateSparseCheckoutFile([string]$mainSpecDir, [string[]]$specAdditionalSubDirectories) {
+    $sparseCheckoutFile = ".git/info/sparse-checkout"
```

Comment:
This isn't used in the underlying function, and I think it's ok to keep hardcoding the path in the underlying function as well.

Also, are we guaranteed that this will always run in a sparse git directory? If not, we should only append to the file if it exists.